service/k8sclient/k8client.go

```diff
@@ -277,8 +281,53 @@ func (c *K8Client) ListXtraDBClusters(ctx context.Context) ([]XtraDBCluster, err
 	return res, nil
 }
 
+// CreateSecret creates secret resource to use as credential source for clusters.
+func (c *K8sClient) CreateSecret(ctx context.Context, secretName string, data map[string][]byte) error {
+	secret := common.Secret{
+		TypeMeta: common.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Secret",
+		},
+		ObjectMeta: common.ObjectMeta{
+			Name: secretName,
+		},
+		Type: common.SecretTypeOpaque,
+		Data: data,
+	}
+	return c.kubeCtl.Apply(ctx, secret)
+}
+
+func randSeq(n int) string {
+	rand.Seed(time.Now().UnixNano())
+	symbols := []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789~=+%^*/()[]{}/!@#$?|")
+	symbolsLen := len(symbols)
+	b := make([]rune, n)
+	for i := range b {
+		b[i] = symbols[rand.Intn(symbolsLen)]
+	}
+	return string(b)
+}
+
 // CreateXtraDBCluster creates Percona XtraDB cluster with provided parameters.
-func (c *K8Client) CreateXtraDBCluster(ctx context.Context, params *XtraDBParams) error {
+func (c *K8sClient) CreateXtraDBCluster(ctx context.Context, params *XtraDBParams) error {
+	secretName := fmt.Sprintf(pxcSecretNameTmpl, params.Name)
+	pwd := randSeq(passwordLength)
+
+	// TODO: add a link to ticket to set random password for all other users.
+	data := map[string][]byte{
+		"root":         []byte(pwd),
+		"xtrabackup":   []byte("backup_password"),
+		"monitor":      []byte("monitor"),
+		"clustercheck": []byte("clustercheckpassword"),
+		"proxyadmin":   []byte("admin_password"),
+		"pmmserver":    []byte("admin"),
```

Comment:
So we are not going to let user change admin password, right? It might bring some issues during testing and usage for users who tests technical preview.
I suggest to get previous secret that we were using earlier and replace hardcoded passwords by values from there.

Reply:
Good point. Implemented.